pyscript.core/docs/README.md

```diff
@@ -0,0 +1,481 @@
+
+
+### Script Features
+
+These are all special, *script* related features, offered by *@pyscript/core* out of the box.
+
+<details>
+  <summary><strong>document.currentScript</strong></summary>
+  <div>
+
+No matter the interpreter of choice, if there is any way to reach the `document` from such interpreter, its `currentScript` will point at the exact/very-same script that is currently executing the code, even if its `async` attribute is used, mimicking what the standard [document.currentScript](https://developer.mozilla.org/en-US/docs/Web/API/Document/currentScript) offers already, and in an unobtrusive way for the rest of the page, as this property only exists for *synchronous* and blocking scripts that are running, hence never interfering with this *core* logic or vice-versa.
+
+```html
+<script type="micropython" id="my-target">
+    from js import document
+
+    # explicitly grab the current script as target
+    my_target = document.getElementById('my-target')
+
+    # verify it is the exact same node with same id
+    print(document.currentScript.id == my_target.id)
+</script>
+```
+
+Not only this is helpful to crawl the surrounding *DOM* or *HTML*, every script will also have a `target` property that will point either to the element reachable through the `target` attribute, or it lazily creates once a companion element that will be appended right after the currently executing *script*.
+
+Please read the [Terminology](#terminology) **target** dedicated details to know more.
+
+  </div>
+</details>
+
+<details>
+  <summary><strong>XWorker</strong></summary>
+  <div>
+
+With or without access to the `document`, every (*non experimental*) interpreter will have defined, at the global level, a reference to the `XWorker` "_class_" (it's just a *function*!), which goal is to enable off-loading heavy operations on a worker, without blocking the main / UI thread (the current page) and allowing such worker to even reach the `document` or anything else available on the very same main / UI thread.
+
+```html
+<script type="micropython">
+    # XWorker is globally defined
+    print(XWorker != None)
+</script>
+```
+
+Please read the [XWorker](#xworker) dedicated section to know more.
+
+  </div>
+</details>
+
+
+## How Events Work
+
+Inspired by the current [HTML Standard](https://html.spec.whatwg.org/multipage/webappapis.html#event-handlers):
+
+> the event handler is exposed through a name, which is a string that always starts with "_on_" and is followed by the name of the event for which the handler is intended.
+
+We took a similar approach, replacing that `on` prefix with whatever *interpreter* or *custom type* is available on the page, plus a *dash* `-` to avoid clashing with standards:
+
+```html
+<script type="micropython">
+    def print_type(event, double):
+        # logs "click 4"
+        print(f"{event.type} {double(2)}")
+</script>
+<button micropython-click="print_type(event, lambda x: x * 2)">
+    print type
+</button>
+```
```

Comment:
Comment here mostly about future..... I think we'll have to work that muscle a lit bit because (I think!!!!) this will be ambiguous. What happens if you have 2 `micropython` enviroments? In that sense we'd need a way to reference an interpreter by type and/or by name... or something like that. 

Not proposing a solution, just highlighting a possible corner case that we need to consider for the future